apps/prairielearn/src/ee/lib/ai-grading.ts

```diff
@@ -275,6 +353,14 @@ export async function aiGrade({
       InstanceQuestionSchema,
     );
 
+    const rubric_items = await queryRows(
+      sql.select_rubric_for_grading,
+      {
+        assessment_question_id: assessment_question.id,
+      },
+      RubricItemSchema,
+    );
```

Comment:
Can this potentially lead to a race condition where the instructor can modify the rubric items after the AI grading job starts? How do we want to approach this?

Reply:
One thing we can do (sacrificing just a little bit of performance) is to do this before generating the gpt prompt, but even this way, worst case, the instructor could change the rubric in between the prompt generation and updating the score with the rubric. Is there a way to help detect this?

Reply:
Pinging @jonatanschroeder and @mwest1066 for suggestions

Reply:
Can you detect that the rubric has changed? If so, I suggest that you throw an error and discard the LLM grading result.

Reply:
I just took a look at the `updateInstanceQuestionScore` function, and it doesn't look like this function is checking if the rubric/rubric item is valid or not? If that is the case, I can do a loop to check if the rubric items have been modified right before updating the instance question score, and throw error if needed.

Reply:
Related: #6802

Reply:
Jonathan, thanks for letting me know. I've implemented a quick check on the rubrics before and after the openai api call to avoid the rubric being changed in between.